src/main/java/org/openstreetmap/atlas/utilities/filters/AtlasEntityPolygonsFilter.java

```diff
@@ -0,0 +1,168 @@
+package org.openstreetmap.atlas.utilities.filters;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.openstreetmap.atlas.geography.Polygon;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.geography.converters.PolygonStringConverter;
+import org.openstreetmap.atlas.geography.converters.WkbPolygonConverter;
+import org.openstreetmap.atlas.geography.converters.WktPolygonConverter;
+import org.openstreetmap.atlas.geography.index.QuadTree;
+import org.openstreetmap.atlas.streaming.readers.GeoJsonReader;
+import org.openstreetmap.atlas.streaming.readers.json.serializers.PropertiesLocated;
+import org.openstreetmap.atlas.streaming.resource.StringResource;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.conversion.StringConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.io.WKBReader;
+
+/**
+ * Configurable {@link AtlasEntity} filter that passes through anything that intersects with include
+ * polygons or anything that doesn't intersect with exclude polygons. Exclude polygons are looked at
+ * only if no include polygons exist. Include takes precedence and polygons intersecting with
+ * previously read polygons are dropped with a warning.
+ * 
+ * @author jklamer
+ */
+public class AtlasEntityPolygonsFilter implements Predicate<AtlasEntity>, Serializable
+{
+
+    public static final String EXCLUDED_POLYGONS_KEY = "filter.polygons.exclude";
+    public static final String INCLUDED_POLYGONS_KEY = "filter.polygons.include";
+    private static final PolygonStringConverter POLYGON_STRING_CONVERTER = new PolygonStringConverter();
+    private static final List<String> POLYGON_STRING_FORMATS = Arrays.asList("atlas", "geojson",
+            "wkt", "wkb");
+    private static final WkbPolygonConverter WKB_POLYGON_CONVERTER = new WkbPolygonConverter();
+    private static final WktPolygonConverter WKT_POLYGON_CONVERTER = new WktPolygonConverter();
+    private static final Logger logger = LoggerFactory.getLogger(AtlasEntityPolygonsFilter.class);
+    private static final long serialVersionUID = -3474748398986569205L;
+    private List<Polygon> exclude = new ArrayList<>();
+    private List<Polygon> include = new ArrayList<>();
+
+    private static StringConverter<Optional<List<Polygon>>> getConverterForFormat(
```

Comment:
Any reason this and the next method are static?

Reply:
They don't rely on any instance variables. Is that a bad design choice?